.github/workflows/python-lint-analyze-test.yml

```diff
@@ -0,0 +1,110 @@
+# Workflow will use Python 3.8 in lint stages using flake8, black and pydocstyle, analyze stage with codeQL, test stage using pytest and build stage using Sphinx
+name: python-lint-analyze-test
+
+on:
+    push:
+        branches:
+            - main
+
+jobs:
+    pre-commit:
+        runs-on: ubuntu-latest
+        strategy:
+            fail-fast: false
+            matrix:
+                python-version: [3.8]
+        steps:
+            - name: Set up creds
+              run: echo "machine github.com login github-ci-token password ${{secrets.GITHUB_TOKEN}}" > ~/.netrc
+
+            - uses: de-vri-es/setup-git-credentials@v2
+              with:
+                  credentials: ${{secrets.GITHUB_TOKEN}}
```

Comment:
What is this needed for? It seems this job will only checkout the local repo and possibly the public repos in the pre-commit config.

Reply:
You are right. This actually helped me catch a badly configured .pre-commit-config.yaml that I had in my other repos. This pipeline has been rectified.